### PR TITLE
Allow skipping private users when grabbing following list

### PR DESCRIPTION
**Pull request by photogsy**
I've found it really useful to be able to grab a list of the people I'm following that are _not_ private. I can then use this list as a starting point which I can filter further, then use for unfollowing.

Very small changes but a really useful (IMHO) feature.
I've been using this for a week or so with expected behaviour.

### CODE REVIEW

**Comment by sionking** [comment]
I like private users :)

**Comment by timgrossmann** [comment]
@photogsy Could you elaborate a little bit more on why this is useful?
What is the reason you don't want to unfollow private users?

Thank you very much for PRing your feature!

**Comment by photogsy** [comment]
Basically its useful to be able to grab a user's following list but ignore private users since you don't want to worry about how many requests to follow you are sending to private accounts.

The problem before was that if you ask InstaPy to grab 500 of a users following list but only want to interact with public accounts, then you end up with far fewer useful users in your list than 500 since you are then going to skip the private ones later.

Hope that makes sense!

I am going to PR a similar change for grabbing a user's followers.

**Comment by timgrossmann** [comment]
@photogsy What is the reason this is called `ignore_private` at one place and `skip_private` at the other?

Also we would need some documentation before merging this. Do you think you could add a quick section about it in the README?

**Comment by timgrossmann** [comment]
@photogsy Any updates on this?

**Comment by timgrossmann** [comment]
@photogsy Any updates on this? 😊

**Comment by photogsy** [comment]
Sorry I haven't had time.
There is no reason for the parameter naming difference, my Python knowledge is limited so I wasn't sure about scope, so to avoid possible issues I named them differently.

**Comment by timgrossmann** [comment]
@photogsy No worries, thank you for the response.

I think it's a good idea to allow something like this, we just need to make sure that the logic is right.
Also, the naming should be unified so it should be called `skip_private` everywhere. 

Thank you very much for proposing this

File: instapy/relationship_tools.py [hunk 1]
```diff
@@ -258,7 +259,7 @@ def get_following(browser,
 
     graphql_endpoint = 'https://www.instagram.com/graphql/query/'
     graphql_following = (
-        graphql_endpoint + '?query_hash=58712303d941c6855d4e888c5f0cd22f')
+        graphql_endpoint + '?query_hash=c56ee0ae1f89cdbd1c89e2bc6b8f3d18')
```
Comment (timgrossmann) [inline instapy/relationship_tools.py:262]: Double check this GraphQL query hash

File: instapy/relationship_tools.py [hunk 2]
```diff
@@ -321,7 +322,8 @@ def get_following(browser,
                 data['user']['edge_follow']['page_info'])
             edges = data['user']['edge_follow']['edges']
             for user in edges:
-                all_following.append(user['node']['username'])
+                if (skip_private == False or user['node']['is_private'] == False):
```
Comment (timgrossmann) [inline instapy/relationship_tools.py:325]: boolean checking in python should be done with `is True/False` instead of `== True/False`

**Review by timgrossmann — CHANGES_REQUESTED** [review]

**Comment by timgrossmann** [comment]
@photogsy Sorry to see that we could not get this features merged.
If at one point you can update this to the latest state of InstaPy and apply the adjustments, we can definitely merge this. I'm sure a lot of users will find this useful.

Thank you very much

**Pull request closed without merging**